packages/task/src/node/process/process-task-runner.ts

```diff
@@ -139,6 +109,73 @@ export class ProcessTaskRunner implements TaskRunner {
         }
     }
 
+    private getResolvedCommand(taskConfig: TaskConfiguration): {
+        command: string | undefined,
+        args: Array<string | QuotedString> | undefined,
+        options: CommandOptions
+    } {
+        let systemSpecificCommand: {
+            command: string | undefined,
+            args: Array<string | QuotedString> | undefined,
+            options: CommandOptions
+        };
+        // on windows, windows-specific options, if available, take precedence
+        if (isWindows && taskConfig.windows !== undefined) {
+            systemSpecificCommand = this.getSystemSpecificCommand(taskConfig, 'windows');
+        } else if (isOSX && taskConfig.osx !== undefined) { // on macOS, mac-specific options, if available, take precedence
+            systemSpecificCommand = this.getSystemSpecificCommand(taskConfig, 'osx');
+        } else { // on linux
+            if (taskConfig.linux !== undefined) { // linux-specific options, if available, take precedence
+                systemSpecificCommand = this.getSystemSpecificCommand(taskConfig, 'linux');
```

Comment:
Unfortunately, I don't have a chance to test it on Windows/MacOS.
Let's assume I have the following task which I'm going run on Windows/MacOS:
```json
		{
			"label": "Run Dev",
			"type": "shell",
			"command": ".\\scripts\\code.bat",
			"linux": {
				"command": "./scripts/code.sh"
			}
		}
```
If I understand it right, Theia would try to execute the command from `linux.command` property. Isn't it?

Reply:
yes, properties that you have in "linux" would override the default values. so in this case `./scripts/code.sh` should be run.

Reply:
if you run that task on Windows/OSX, ".\\scripts\\code.bat" should be run. `linux.command` is only for execution on linux.
please correct me if my code is wrong.

Reply:
hmm, i see what you mean - there is a bug with the if - else logic here. let me do more testing and fix it.